service/src/stock_take/delete.rs

```
@@ -0,0 +1,53 @@
+use repository::{RepositoryError, StockTakeRowRepository, StorageConnection, TransactionError};
+
+use crate::{service_provider::ServiceContext, validate::check_store_id_matches};
+
+use super::validate::check_stock_take_exist;
+
+#[derive(Debug, PartialEq)]
+pub enum DeleteStockTakeError {
+    DatabaseError(RepositoryError),
+    StockTakeDoesNotExist,
```

stock take is not empty (i.e. has lines)

---

thanks done in b1964fe (I leave the test for another PR because I already have stock line mock data in a different branch) test is now also failing because of it will fix in next iteration